crates/matrix-sdk-crypto/src/backups/keys/compat.rs

```diff
@@ -207,6 +207,13 @@ pub enum Error {
     /// The message failed to be decoded.
     #[error("The message could not been decoded: {0}")]
     Decoding(#[from] MessageDecodeError),
+    /// The message's Curve25519 key failed to be decoded.
+    #[error("The message's ephemeral Curve25519 key could not been decoded: {0}")]
+    InvalidCurveKey(#[from] KeyError),
+    /// The decrypted message should contain a backed up room key, but the
+    /// plaintext isn't valid JSON.
+    #[error("The decrypted message isn't valid JSON: {0}")]
+    Json(#[from] serde_json::error::Error),
```

Comment:
(Probably not actionable, but...) For those kinds of invalid JSON errors, I always would like to have the piece of the JSON object that's causing the error, and maybe the JSON path to the object/array entry, so as to make it easier to debug. Is that something that we can enable easily?

Reply:
There were talks that we're going to start using a crate for this, AFAIR it was [`serde_path_to_error`](https://docs.rs/serde_path_to_error/latest/serde_path_to_error/). Not sure what happened to that idea.